### ISSUE TITLE
Git Resource (PipelineResource) seems to be broken for GitHub LFS

### ISSUE THREAD

**Issue by eqqe** [tektoncd/pipeline#2997]
# Expected Behavior
A Git Resource pointing to a GitHub using LFS should be cloned correctly.
# Actual Behavior
The large files are not cloned correctly they are replaced by small files (100 Bytes instead of 150MB)

# Steps to Reproduce the Problem

1. Create a GitHub repository and push large files using Git LFS
2. Create a Git Resource (PipelineResource) pointing to this GitHub repository
3. Launch the pipeline and see that the message "Successfully cloned" and notice that the files size is incorrect.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:52:00Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.3", GitCommit:"2e7996e3e2712684bc73f0dec0200d64eec7fe40", GitTreeState:"clean", BuildDate:"2020-05-20T12:43:34Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
  ```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**
v0.14.2

**Comment by sbwsg** [comment]
I hadn't heard of git lfs before, for anyone else looking at this it's documented here: https://git-lfs.github.com/

Sounds like we'd need to install the lfs extension as part of tekton's git-init image.

**Comment by vdemeester** [comment]
@eqqe @sbwsg indeed, we do not have `git-lfs` installed in our `git-init` image so we don't support Git LFS. As PipelineResource are not in `v1beta1` and their future still being discussed, I think the best "workaround" right now would be to use a `git-clone` task (like the [one in the catalog](https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1) but with a different image/script :sweat_smile: ).

**Comment by eqqe** [comment]
Opened PR #3006

**Comment by sbwsg** [comment]
Looks like this has been resolved by #3006 which was in the 0.16 release.
